### PR TITLE
chore(config): Make config schema output ordered

### PR DESCRIPTION
**Pull request by bruceg**
Key/value maps in the config schema currently use an `IndexMap`. This provides for stable ordering of the output between runs. Unfortunately, due to the use of initializers via `inventory`, that ordering is dependent on the order in which those initializers are run, which is in turn dependent on the choices the linker (and link-time optimization) makes when building Vector. The result is that config schemas can be very hard to compare between builds due to large blocks of it moving around in the output file.

This change changes this type to a `BTreeMap`, which is fully ordered internally. This will result in a consistent ordering of maps, hopefully producing config schema output that is more consistent.

### CODE REVIEW

File: regression/Dockerfile
Comment (lukesteensen) [inline regression/Dockerfile:1]: Is this change related?

**Review by lukesteensen — APPROVED** [review]
Looks reasonable to me

File: regression/Dockerfile
Comment (bruceg) [reply regression/Dockerfile:1]: No, that was unrelated, I mistakenly committed everything I had in my working copy.
